lib/datadog/core/telemetry/logger.rb

```diff
@@ -8,7 +8,7 @@ module Telemetry
       # Report telemetry logs via delegating to the telemetry component instance via mutex.
       #
       # IMPORTANT: Invoking this method during the lifecycle of component initialization will
-      # cause a non-recoverable deadlock
+      # be no-op instead.
```

Comment:
While causing a deadlock is really bad, I think we should plan to fix the underlying problem. E.g. I think the core should provide to components the service of making telemetry available very early on; it seems reasonable for components to want to report data during initialization.